File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l = 0, r = numbers.length - 1;

        while(l < r){
            int sum = numbers[l] + numbers[r];
            // if current sum greater than target -> move right to mid
            if (sum > target){
                r--;
            }
            // elif current sum less than target -> move left to mid
            else if (sum < target){
                l++;
            }
            // else sum equals target -> return those indexes + 1
            else return new int[]{l+1, r+1};
        }
        return new int[0];
    }
}

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        return self.dfs(nums, target, 0, len(nums) - 1)
        
    def dfs(self, nums, target, l, r):
        if l > r:
            return -1

        m = l + (r - l) // 2
        if nums[m] == target:
            return m
        elif nums[m] < target:
            return self.dfs(nums, target, l + 1, r)
        else:
            return self.dfs(nums, target, l, r - 1)
    
